tests/test_fixity.py

```diff
@@ -634,3 +635,58 @@ def test_scanall_if_sort_argument_is_passed(
             params={"username": STORAGE_SERVICE_USER, "api_key": STORAGE_SERVICE_KEY},
         ),
     ]
+
+
+@mock.patch("requests.get")
+def test_main_handles_exception_if_environment_key_is_missing(
+    _get: mock.Mock, mock_check_fixity: List[mock.Mock]
+) -> None:
+    _get.side_effect = mock_check_fixity
+    aip_id = uuid.uuid4()
+    stream = io.StringIO()
+
+    fixity.main(["scan", str(aip_id)], stream=stream)
+
+    assert fixity.ArgumentError("Missing environment variable: STORAGE_SERVICE_URL")
+
+
+@mock.patch("requests.get")
+def test_scanall_handles_exception_if_storage_service_is_not_connected(
+    _get: mock.Mock, monkeypatch: pytest.MonkeyPatch
+) -> None:
+    monkeypatch.setenv("STORAGE_SERVICE_URL", STORAGE_SERVICE_URL)
+    monkeypatch.setenv("STORAGE_SERVICE_USER", "")
+    monkeypatch.setenv("STORAGE_SERVICE_KEY", "")
+    aip_id1 = str(uuid.uuid4())
+    aip_id2 = str(uuid.uuid4())
+    _get.side_effect = [
+        mock.Mock(
+            **{
+                "status_code": 401,
+                "json.return_value": {
+                    "meta": {"next": None},
+                    "objects": [
+                        {
+                            "package_type": "AIP",
+                            "status": "UPLOADED",
+                            "uuid": f"{aip_id1}",
+                        },
+                        {
+                            "package_type": "AIP",
+                            "status": "UPLOADED",
+                            "uuid": f"{aip_id2}",
+                        },
+                    ],
+                },
+            },
+            spec=requests.Response,
+            side_effect=ConnectionError,
+        )
+    ]
+    stream = io.StringIO()
+
+    fixity.main(["scanall"], stream=stream)
+
+    assert storage_service.StorageServiceError(  # type: ignore
+        f'Storage service at "{STORAGE_SERVICE_URL}" failed authentication while requesting AIPs'
+    )
```

Comment:
Could you explain what is the goal of this assertion? Also, why does it need to be ignored?

Reply:
An exception raises when there is connection error from storage service.

Reply:
So, if I understand you correctly, you're looking to create a connection error scenario in the test and confirm that you got one? If that's the case I don't think this `assert` is accomplishing that.

You could try this very simple exercise: remember our discussions of setting up tests in a BDD-style of ["given, when, then"](https://martinfowler.com/bliki/GivenWhenThen.html)? Remove the body of the test except for the `assert` statement (i.e. the "given" and "when" parts). Your test still passes, right? That demonstrates that the assertion (i.e. the "then") doesn't depend on the scenario at all.

Reply:
My main focus is to raise `StorageServiceError` . I guess I should change the test name since it is not related with `ConnectionError` with storage service.